### PR TITLE
Editorial: fix requirement of opening a new path

### PR DESCRIPTION
**Pull request by Yanmei-Liu**
Fix issue #407

### CODE REVIEW

**Review by huitema — CHANGES_REQUESTED** [review]
I don't think this change is necessary. It is also not purely editorial. We discussed that before. We recommend that NEW_CONNECTION_ID frames be sent in order of increasing path ID, but we can never guarantee that they will be *received* in order. Suppose the server send (C1, pathID 1) and (C2, pathID 2), and that C1 is delayed, maybe by packet loss. Your new text will say that the client MUST wait until C1 is received to open a new path. That's wrong.

Also, it is better to have all the discussion of order, etc., in exactly one place. The previous text is better.

**Comment by Yanmei-Liu** [comment]
> I don't think this change is necessary. It is also not purely editorial. We discussed that before. We recommend that NEW_CONNECTION_ID frames be sent in order of increasing path ID, but we can never guarantee that they will be _received_ in order. Suppose the server send (C1, pathID 1) and (C2, pathID 2), and that C1 is delayed, maybe by packet loss. Your new text will say that the client MUST wait until C1 is received to open a new path. That's wrong.
> 
> Also, it is better to have all the discussion of order, etc., in exactly one place. The previous text is better.

@huitema But in [Section 3.5](https://datatracker.ietf.org/doc/html/draft-ietf-quic-multipath-10#section-3.5) we use "Similarly, endpoints SHOULD consume Path IDs in a continuous way, i.e., when creating paths." I think in the case of retransmission, endpoints should choose to wait for C1 to create the new path, using C2 would break the "SHOULD". 

BTW, this issue and pull request is to solve the true requirement of new path initialization is the use of new Path ID. I changed the description of "the smallest unused Path ID" to "an unused Path ID". Please check again.

**Review by mirjak — APPROVED** [review]

File: draft-ietf-quic-multipath.md [hunk 1]
```diff
@@ -363,8 +363,8 @@ All new frames are sent in 1-RTT packets {{QUIC-TRANSPORT}}.
 
 ## Path Initiation {#path-initiation}
 
-Opening a new path requires the
-use of a new connection ID (see {{Section 9.5 of QUIC-TRANSPORT}}).
+Opening a new path requires the use of a new connection ID
+for an unused Path ID (see {{explicit-path-id}}).
```
Comment (mirjak) [inline draft-ietf-quic-multipath.md:367]:
```suggestion
Opening a new path requires the use of a new connection ID (see {{Section 9.5 of QUIC-TRANSPORT}})
for an unused Path ID (see {{explicit-path-id}}).
```
I think it is actually not correct to remove the reference to section 9.5 of RFC9000 as that section explain the privacy implications and we don't have a reference to 9.5 of RFC9000 in the section on explicit path ID. However, I think we can have both references.

**Review by mirjak — CHANGES_REQUESTED** [review]
I think we still need the reference to sec 9.5 of RFC9000

File: draft-ietf-quic-multipath.md [hunk 2]
```diff
@@ -363,8 +363,8 @@ All new frames are sent in 1-RTT packets {{QUIC-TRANSPORT}}.
 
 ## Path Initiation {#path-initiation}
 
-Opening a new path requires the
-use of a new connection ID (see {{Section 9.5 of QUIC-TRANSPORT}}).
+Opening a new path requires the use of a new connection ID (see {{Section 9.5 of QUIC-TRANSPORT}})
+for an unused Path ID (see {{explicit-path-id}}).
```
Comment (qdeconinck) [inline draft-ietf-quic-multipath.md:367]:
```suggestion
mapped to an unused Path ID (see {{explicit-path-id}}).
```

**Review by qdeconinck — APPROVED** [review]
small suggestion

**Review by huitema — APPROVED** [review]

**Pull request merged**